File: Medium/Python/Math/MySqrt.py
class Solution(object):
    def mySqrt(self, x):
        """
            :type x: int
            :rtype: int
            """
        if x == 1:
            return 1
        temp = x / 2
        while temp ** 2 > x:
            temp = (temp + x / temp) / 2
        return temp

File: Medium/Java/Math/MySqrt.java
class Solution {
    public int mySqrt(int x) {
        if (x <= 1)
            return x;
        int temp = x / 2;
        while (Math.pow(temp, 2) > x)
            temp = (temp + x / temp) / 2;
        return temp;
    }
}
